base/iostream.jl

```diff
@@ -264,7 +264,7 @@ end
 function readbytes_all!(s::IOStream, b::Array{UInt8}, nb)
     olb = lb = length(b)
     nr = 0
-    while nr < nb
+    @gc_preserve b while nr < nb
```

Comment:
should we just use `Ref(b, nr + 1)` instead?

Reply:
There's a few different places that uses `pointer(ary, i)` that can be changed, yes.